aws/ec2/monitorInstances.go

```diff
@@ -0,0 +1,284 @@
+		logger.Error("Error when describing instances", err.Error())
+		return
+	}
+	start, end := getCurrentCheckedHour()
+	for _, reservation := range instances.Reservations {
+		for _, instance := range reservation.Instances {
+			CpuAverage, CpuPeak := getInstanceStats(ctx, aws.StringValue(instance.InstanceId), sess, region)
+			instanceInfoChan <- InstanceInfo{
+				Account:    account,
+				Service:    "EC2",
+				Id:         aws.StringValue(instance.InstanceId),
+				Region:     aws.StringValue(instance.Placement.AvailabilityZone),
+				StartDate:  start,
+				EndDate:    end,
+				KeyPair:    aws.StringValue(instance.KeyName),
+				Tags:       getInstanceTag(instance.Tags),
+				Type:       aws.StringValue(instance.InstanceType),
+				CpuAverage: CpuAverage,
+				CpuPeak:    CpuPeak,
+			}
+		}
+	}
+}
+
+// fetchRegionsList fetchs the regions list from AWS and returns an array of their name.
+func fetchRegionsList(ctx context.Context, sess *session.Session) ([]string, error) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	svc := ec2.New(sess)
+	regions, err := svc.DescribeRegions(nil)
+	if err != nil {
+		logger.Error("Error when describing regions", err.Error())
+		return []string{}, err
+	}
+	res := make([]string, 0)
+	for _, region := range regions.Regions {
+		res = append(res, aws.StringValue(region.RegionName))
+	}
+	return res, nil
+}
+
+// createIndexEs creates the ElasticSearch index with the properties
+// from getPropertiesMapping.
+func createIndexEs(ctx context.Context, client *elastic.Client) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	mapping := "{\"mappings\":{\"" + DocumentEc2 + "\":" + mappingEc2Instance + "}}"
+	if _, err := client.CreateIndex(IndexHourlyInstanceUsage).BodyString(mapping).Do(context.Background()); err != nil {
+		logger.Error("Error when creating ElasticSearch index", err.Error())
+	}
+}
+
+// importInstancesToEs imports an array of InstanceInfo in ElasticSearch.
+// It calls createIndexEs if the index doesn't exist.
+func importInstancesToEs(ctx context.Context, instances []InstanceInfo) {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	client := es.Client
+	if exists, err := client.IndexExists(IndexHourlyInstanceUsage).Do(context.Background()); err != nil {
+		logger.Error("Error when checking if the ElasticSearch index exists", err.Error())
+		return
+	} else if !exists {
+		createIndexEs(ctx, client)
+	}
+	for _, instance := range instances {
+		ji, err := json.Marshal(instance)
+		if err != nil {
+			logger.Error("Error when marshaling instance var", err.Error())
+			continue
+		}
+		hash := md5.Sum(ji)
+		hash64 := base64.URLEncoding.EncodeToString(hash[:])
+		if res, err := client.
+			Index().
+			Index(IndexHourlyInstanceUsage).
+			Type(DocumentEc2).
+			BodyJson(instance).
+			Id(hash64).
+			Do(context.Background()); err != nil {
+			logger.Error("Error when putting InstanceInfo in ES", err.Error())
+		} else {
+			logger.Info("Instance put in ES", *res)
+		}
+	}
```

Comment:
This function does again not return anything while function called inside of it can return errors.
Since you assume that this function will always succeed, what happens and it fails silently ? Will this be a problem for other functions depending on it ?